core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -384,15 +382,15 @@ func (ec *EthConfirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE])
 	}
 
 	for from, attempts := range attemptsByAddress {
-		minedTransactionCount, err := ec.getMinedTransactionCount(ctx, from)
+		minedTransactionCount, err := ec.getMinedSequenceForAddress(ctx, from)
 		if err != nil {
 			return errors.Wrapf(err, "unable to fetch pending nonce for address: %v", from)
 		}
 
 		// separateLikelyConfirmedAttempts is used as an optimisation: there is
 		// no point trying to fetch receipts for attempts with a nonce higher
 		// than the highest nonce the RPC node thinks it has seen
-		likelyConfirmed := ec.separateLikelyConfirmedAttempts(from, attempts, minedTransactionCount)
+		likelyConfirmed := ec.separateLikelyConfirmedAttempts(from, attempts, uint64(minedTransactionCount.Int64()))
```

Comment:
Can this be converted directly?
```suggestion
		likelyConfirmed := ec.separateLikelyConfirmedAttempts(from, attempts, uint64(minedTransactionCount))
```

Reply:
Yep, good point 👍 
I was also thinking if Nonce should be uint64 so that we don't cast it every time. 
@prashantkumar1982 was there particular reason Nonce is int64?

Reply:
For some reason our DB schema uses bigint, which is int64. And that's how then we represent it as int64 in our in-memory data structures.
I see geth uses uint64.
We also use  at a place, lastMinedNonce. For a brand new Address, that value is -1. Maybe that's why we had to choose int64.